apps/explorer/src/ui/SplitPanes.tsx

```diff
@@ -26,16 +39,95 @@ function ResizeHandle({ isHorizontal }: { isHorizontal: boolean }) {
             <div
                 data-is-dragging={isDragging}
                 className={clsx(
-                    'bg-gray-45 group-hover:bg-sui data-[is-dragging=true]:bg-hero',
+                    'relative bg-gray-45 group-hover:bg-sui data-[is-dragging=true]:bg-hero',
                     isHorizontal ? 'h-full w-px' : 'h-px'
                 )}
-            />
+            >
+                {collapsibleButton && (
+                    <div
+                        role="button"
+                        onClick={togglePanelCollapse}
+                        data-is-dragging={isDragging}
+                        className={clsx([
+                            'flex cursor-pointer items-center rounded-full',
+                            'border border-gray-45 bg-white text-gray-70',
+                            'hover:bg-sui hover:text-white group-hover:block',
+                            isCollapsed ? 'block' : 'hidden',
+                            isHorizontal
+                                ? 'absolute left-1/2 top-[5%] -translate-x-2/4'
+                                : 'absolute left-[5%] top-1/2 -translate-y-2/4',
+                        ])}
+                    >
+                        <ChevronButton
+                            height={16}
+                            width={16}
+                            className={clsx(isCollapsed && 'rotate-180')}
+                        />
+                    </div>
+                )}
+            </div>
         </PanelResizeHandle>
     );
 }
 
+interface SplitPanelProps extends PanelProps {
+    panel: ReactNode;
+    direction: Direction;
+    renderResizeHandle: boolean;
+    collapsibleButton?: boolean;
+}
+
+function SplitPanel({
+    panel,
+    direction,
+    renderResizeHandle,
+    collapsibleButton,
+    ...props
+}: SplitPanelProps) {
+    const ref = useRef<ImperativePanelHandle>(null);
+    const [isCollapsed, setIsCollapsed] = useState(false);
+
+    const togglePanelCollapse = () => {
```

Comment:
Might be a little cleaner to do this inside a useEffect and just call the setter in the child